go/private/extensions.bzl

```diff
@@ -1,6 +1,24 @@
-load("//go/private:sdk.bzl", "go_download_sdk_rule", "go_host_sdk_rule", "go_multiple_toolchains")
+load("//go/private:sdk.bzl", "detect_host_platform", "go_download_sdk_rule", "go_host_sdk_rule", "go_multiple_toolchains")
 load("//go/private:repositories.bzl", "go_rules_dependencies")
 
+def host_compatible_toolchain_impl(ctx):
+    ctx.file("BUILD.bazel", content = "")
+    host_compatible_sdk = ctx.attr.toolchains[0]
+    ctx.file("defs.bzl", content = """
+host_compatible_sdk = Label("{}")
+""".format(host_compatible_sdk))
+
+host_compatible_toolchain = repository_rule(
+    implementation = host_compatible_toolchain_impl,
+    attrs = {
+        "toolchains": attr.string_list(
```

Comment:
Since we only use the first toolchain anyway, could you make this an `attr.string` instead?

Could you also add a comment explaining that we can't use `attr.label` since the module extension itself can't refer to the repositories it creates by their apparent repository names?